projects/kit/components/calendar-range/calendar-range.component.ts

```diff
@@ -170,7 +170,10 @@ export class TuiCalendarRange implements OnInit, OnChanges {
         ...items.filter(
             (item) =>
                 (minLength === null ||
-                    item.range.from.append(minLength).daySameOrBefore(item.range.to)) &&
+                    item.range.from
+                        .append(minLength)
+                        .append({day: -1})
+                        .daySameOrBefore(item.range.to)) &&
```

Comment:
@mdlufy Can we implement tests for this case?

Reply:
Yes, it would be good to add a test,  I think we can do it

Reply:
@bondarvladislave do you have a desire to add a test?)